synapse/handlers/devicemessage.py

```diff
@@ -153,6 +161,14 @@ async def send_device_message(
         local_messages = {}
         remote_messages = {}  # type: Dict[str, Dict[str, Dict[str, JsonDict]]]
         for user_id, by_device in messages.items():
+            # Ratelimit local cross-user key requests by user/device.
+            if (
+                message_type == EventTypes.RoomKeyRequest
+                and user_id != sender_user_id
+                and self._ratelimiter.can_do_action((user_id, by_device))
```

Comment:
I wasn't sure about using this as the key, but it seemed reasonable?

Reply:
Isn't `by_device` a dict? I think it should probably be the sender device ID, which isn't passed in here but can be easily added as this only gets called in one place?